TRexLib.InteractiveExtension/PlotlyChartExtensions.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Text;
+using Microsoft.AspNetCore.Html;
+using XPlot.Plotly;
+using static Microsoft.DotNet.Interactive.Formatting.PocketViewTags;
+
+namespace TRexLib.InteractiveExtension
+{
+    public static class PlotlyChartExtensions
+    {
+        public static IHtmlContent GetHtmlContent(this PlotlyChart chart)
+        {
+            var divElement = div[style: $"width: {chart.Width}px; height: {chart.Height}px;", id: chart.Id]();
+            var jsElement = chart.GetInlineJS().Replace("<script>", string.Empty).Replace("</script>", string.Empty);
+
+            return new HtmlString($@"{divElement}
+{GetScriptElementWithRequire(jsElement)}");
+        }
+
+        private static IHtmlContent GetScriptElementWithRequire(string script)
+        {
+            var newScript = new StringBuilder();
+            newScript.AppendLine("<script type=\"text/javascript\">");
+            newScript.AppendLine(@"
+var renderPlotly = function() {
```

Comment:
How come you need to do this here? Why doesn't the built-in Plotly support take care of it?

Reply:
The built-in support is in a library that we haven't published and probably don't want to publish independently because it should move into `Xplot.Plotly`, ideally. And the `GetHtml` methods in `XPlot.Plotly` don't emit the script references. So overall, this was simpler and had fewer dependencies.